addons/im_livechat/models/mail_message_reaction.py

```diff
@@ -0,0 +1,21 @@
+from odoo import models
+from odoo.addons.mail.tools.discuss import Store
+
+
+class MailMessageReaction(models.Model):
+    _name = "mail.message.reaction"
+    _inherit = ["mail.message.reaction"]
+
+    def _reacting_persona_to_store(self, store: Store):
+        channel_reactions_with_partner = self.filtered(
+            lambda r: r.partner_id and r.message_id.model == "discuss.channel"
+        )
+        channel_by_message = channel_reactions_with_partner.message_id._record_by_message()
+        livechat_agent_reactions = channel_reactions_with_partner.filtered(
```

Comment:
agent in name is a bit misleading as customer will be taken into account too, only they will be "ignored" later if they don't have a `user_livechat_username`